org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/RenameHandlerTest.java

```diff
@@ -0,0 +1,221 @@
+	}
+
+	@Test
+	public void testRename_localVariable() throws JavaModelException {
+		IPackageFragment pack1 = sourceFolder.createPackageFragment("test1", false, null);
+
+		String[] codes = {
+				"package test1;\n",
+				"public class E {\n",
+				"   public int bar() {\n",
+				"		String str = new String();\n",
+				"   	str.length();\n",
+				"   }\n",
+				"   public int foo() {\n",
+				"		String str = new String();\n",
+				"   	str|*.length()\n",
+				"   }\n",
+				"}\n"
+		};
+		StringBuilder builder = new StringBuilder();
+		Position pos = mergeCode(builder, codes);
+		ICompilationUnit cu = pack1.createCompilationUnit("E.java", builder.toString(), false, null);
+
+		WorkspaceEdit edit = getRenameEdit(cu, pos, "newname");
+		assertNotNull(edit);
+		assertEquals(edit.getChanges().size(), 1);
+		assertEquals(edit.getChanges().get(JDTUtils.getFileURI(cu)).size(), 2);
+	}
+
+	@Test
+	public void testRename_method() throws JavaModelException {
+		IPackageFragment pack1 = sourceFolder.createPackageFragment("test1", false, null);
+
+		String[] codes = {
+				"package test1;\n",
+				"public class E {\n",
+				"   public int bar() {\n",
+				"   }\n",
+				"   public int foo() {\n",
+				"		this.bar|*();\n",
+				"   }\n",
+				"}\n"
+		};
+		StringBuilder builder = new StringBuilder();
+		Position pos = mergeCode(builder, codes);
+		ICompilationUnit cu = pack1.createCompilationUnit("E.java", builder.toString(), false, null);
+
+		WorkspaceEdit edit = getRenameEdit(cu, pos, "newname");
+		assertNotNull(edit);
+		assertEquals(edit.getChanges().size(), 1);
+		assertEquals(edit.getChanges().get(JDTUtils.getFileURI(cu)).size(), 2);
+	}
+
+	@Test
+	public void testRename_systemLibrary() throws JavaModelException {
+		IPackageFragment pack1 = sourceFolder.createPackageFragment("test1", false, null);
+
+		String[] codes = {
+				"package test1;\n",
+				"public class E {\n",
+				"   public int bar() {\n",
+				"		String str = new String();\n",
+				"   	str.len|*gth();\n",
+				"   }\n",
+				"}\n"
+		};
+		StringBuilder builder = new StringBuilder();
+		Position pos = mergeCode(builder, codes);
+		ICompilationUnit cu = pack1.createCompilationUnit("E.java", builder.toString(), false, null);
+
+		WorkspaceEdit edit = getRenameEdit(cu, pos, "newname");
+		assertNotNull(edit);
+		assertEquals(edit.getChanges().size(), 0);
+	}
+
+	@Test
+	public void testRename_multipleFiles() throws JavaModelException {
+		IPackageFragment pack1 = sourceFolder.createPackageFragment("test1", false, null);
+
+		String[] codes1= {
+				"package test1;\n",
+				"public class A {\n",
+				"   public void bar() {\n",
+				"		String str = new String();\n",
+				"   	str.length();\n",
+				"   }\n",
+				"   public void foo() {\n",
+				"		String str = new String();\n",
+				"   	str.length()\n",
+				"   }\n",
+				"}\n"
+		};
+
+		String[] codes2 = {
+				"package test1;\n",
+				"public class B {\n",
+				"   public void bar() {\n",
+				"		String str = new String();\n",

```

Comment:
keep the tests minimal and easy to understand and remove the methods and method bodies that are not involved